overlay/libexec/k3os/boot

```diff
@@ -16,7 +16,13 @@ setup_ttys()
             ;;
         console=*)
             CONSOLE_SPEC=${x#console=}
-            IFS=, read TTY BAUDRATE _ <<<"${CONSOLE_SPEC},9600"
+            IFS=, read TTY BAUDRATE _ <<<"${CONSOLE_SPEC}"
+
+            if [ -n "$BAUDRATE" ]; then
+                BAUDRATE=$(grep -o '^[0-9]*' <<< "$BAUDRATE")
```

Comment:
Is this restricting BAUDRATE to be only numbers? If so, I think this should support stuff like `9600n8` (like described in the kernel page you've linked).

Reply:
The reason for only matching the numbers is that getty only expects numbers as the parameter where `$BAUDRATE` is used. As far as I can tell from the manual, getty is able to infer stop bits, checksum etc.

Reply:
AFICT the serial port has no auto-negotiation feature, so I'm now wondering how it does that magic!

Reply:
Well, getty only has a flag for "assume 8bit clean", it doesn't have a flag for 7bit with odd, even, none or space parity, so maybe it guesses and things just work. Anyway, the patch still makes things better, even if it assumes 8bit. In my case I was using a machine with coreboot hard-coded to 115200 baud, meaning I had to start my term i with the baud rate, and switch when the kernel started to see a complete boot log.

Reply:
Indeed it does, no doubt about that :-)

Coreboot? Woah! What machine do you have? I've very curious who is using coreboot :-)

Reply:
It's an https://pcengines.ch/apu2c4.htm, it's a very nice machine I can only recommend them :-)

Reply:
Nice! I've been looking at the apu4d4 model, but I still can't justify getting one eheh